compiler-plugin/src/main/kotlin/arrow/meta/internal/annotation/MetaAnnotations.kt

```diff
@@ -0,0 +1,7 @@
+package arrow
+
+@Retention(AnnotationRetention.SOURCE)
+@Target(AnnotationTarget.EXPRESSION, AnnotationTarget.CLASS, AnnotationTarget.TYPEALIAS,
+  AnnotationTarget.FIELD, AnnotationTarget.CONSTRUCTOR, AnnotationTarget.LOCAL_VARIABLE,
+  AnnotationTarget.FUNCTION)
```

Comment:
@raulraja can we extend this annotation to Types and TypeParameters or is this not an intended behavior?

Reply:
Types and TypeParameter are never asked as synthetic descriptors but it does not hurt to be able to annotate them